src/core/config/tmf.h

```diff
@@ -35,6 +35,8 @@
 #ifndef CONFIG_TMF_H_
 #define CONFIG_TMF_H_
 
+#include "config/device_type_check.h"
```

Comment:
Is it still necessary after we have included "config/device_type_check.h" at the beginning of `openthread-core-config.h`?